python/cuml/test/test_pca.py

```diff
@@ -80,21 +114,38 @@ def test_pca_fit_transform(datatype, input_type):
 
 @pytest.mark.parametrize('datatype', [np.float32, np.float64])
 @pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
-def test_pca_inverse_transform(datatype, input_type):
-    gdf = cudf.DataFrame()
-    gdf['0'] = np.asarray([-1, -2, -3, 1, 2, 3], dtype=datatype)
-    gdf['1'] = np.asarray([-1, -1, -2, 1, 1, 2], dtype=datatype)
+@pytest.mark.parametrize('name', [pytest.param('None', marks=pytest.mark.unit),
+                                  pytest.param('blobs',
+                                               marks=pytest.mark.stress),
+                                  pytest.param('iris',
+                                               marks=pytest.mark.quality)])
+def test_pca_inverse_transform(datatype, input_type,
+                               name):
+    if name == 'blobs':
+        pytest.skip('fails when using blobs dataset')
+        X, y = make_blobs(n_samples=500000,
+                          n_features=1000, random_state=0)
+
+    elif name == 'iris':
+        iris = datasets.load_iris()
+        X = iris.data
+
+    else:
+        X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]],
+                     dtype=datatype)
+
+    X_pd = pd.DataFrame(
+                       {'fea%d' % i: X[0:, i] for i in range(X.shape[1])})
+    X_cudf = cudf.DataFrame.from_pandas(X_pd)
+
     cupca = cuPCA(n_components=2)
 
     if input_type == 'dataframe':
-        Xcupca = cupca.fit_transform(gdf)
+        Xcupca = cupca.fit_transform(X_cudf)
 
     else:
-        X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]],
-                     dtype=datatype)
         Xcupca = cupca.fit_transform(X)
 
     input_gdf = cupca.inverse_transform(Xcupca)
-
-    assert array_equal(input_gdf, gdf,
-                       1e-3, with_sign=True)
+    assert array_equal(input_gdf, X_cudf,
```

Comment:
Is this doable with no floating point error at all? Interesting, I would've expected to need some slight tolerance here.

Reply:
The issue is that with floating point values it gives an assertion error since pca inverse transform does not work well with the iris dataset.